File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
      
        int maxi = 0, mini = prices[0];
        for( int i = 1 ; i < prices.size() ; i++ ){
            int cost = prices[i] - mini;
            maxi = max(maxi,cost);
            mini = min(prices[i],mini);
        }
        return maxi;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxSum = 0 ;
        int stockA = prices[0] ;
        for(int i = 1 ; i < prices.length ; i++){
            if( prices[i] - stockA < 0){
                stockA = prices[i] ;
            } 
            else{
                maxSum = Math.max(maxSum,prices[i]-stockA);
            }

        }
        return maxSum ;
    }
}